node/src/components/linear_chain_sync/state.rs

```diff
@@ -24,9 +24,8 @@ pub enum State {
         /// The most recent block we started to execute. This is updated whenever we start
         /// downloading deploys for the next block to be executed.
         latest_block: Box<Option<Block>>,
-        /// Switch block of the current era.
-        /// Updated whenever we see a new switch block.
-        maybe_switch_block: Option<Box<Block>>,
+        /// The block height of the last seen switch block.
+        last_switch_block_height: Option<u64>,
```

Comment:
Is `BlockHeight` available here?

Reply:
It's defined in `block_proposer`, `contract_runtime` and `effect::requests`, so currently we'd have to define it a fourth time here :grimacing: We should probably define it in a single place and import it from there in others, but that sounds like a separate ticket.